### PR TITLE
processing data in models

### PR DESCRIPTION
**Pull request by maen-al-khraisha**
- Create model classes for all entities as shown in the ERD model.
- Set up associations between models.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,7 @@
 class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id'
```
Comment (rustam-tolipov) [inline app/models/user.rb:2]: - [ ] The user not only has posts he also has `comments`, and `likes` as well. Kindly add these two alongside posts.

File: app/models/post.rb [hunk 2]
```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
```
Comment (rustam-tolipov) [inline app/models/post.rb:2]: - [ ] Its correct posts belong to the user. Also, posts have `comments`, and `likes` as well.

File: app/models/post.rb [hunk 3]
```diff
@@ -1,2 +1,13 @@
...
+
+  after_save :update_user_posts_count
+
+  def update_user_posts_count
```
Comment (rustam-tolipov) [inline app/models/post.rb:8]: [OPTIONAL] I would suggest putting the `update_user_posts_count` `method` after the private method.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @maenkhraisha, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @maenkhraisha  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Linter checks passing ✔️ 
- Made changes as requested by the previous reviewer 👍 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**